test/interp/basic-logging.txt

```diff
@@ -45,50 +45,28 @@
 0000025: 0b                                        ; end
 0000020: 05                                        ; FIXUP func body size
 000001e: 07                                        ; FIXUP section size
-begin_module(1)
-begin_signature_section(5)
-  on_signature_count(1)
-  on_signature(index: 0, params: [], results: [i32])
-end_signature_section
-begin_function_signatures_section(2)
-  on_function_signatures_count(1)
-  on_function_signature(index: 0, sig_index: 0)
-end_function_signatures_section
-begin_export_section(8)
-  on_export_count(1)
-  on_export(index: 0, kind: func, item_index: 0, name: "main")
-end_export_section
-begin_function_bodies_section(7)
-  on_function_bodies_count(1)
-  begin_function_body(0)
-  on_local_decl_count(0)
-  on_i32_const_expr(42 (0x2a))
-  on_return_expr
-  end_function_body(0)
-end_function_bodies_section
-end_module
-begin_module(1)
-begin_signature_section(5)
-  on_signature_count(1)
-  on_signature(index: 0, params: [], results: [i32])
-end_signature_section
-begin_function_signatures_section(2)
-  on_function_signatures_count(1)
-  on_function_signature(index: 0, sig_index: 0)
-end_function_signatures_section
-begin_export_section(8)
-  on_export_count(1)
-  on_export(index: 0, kind: func, item_index: 0, name: "main")
-end_export_section
-begin_function_bodies_section(7)
-  on_function_bodies_count(1)
-  begin_function_body(0)
-  on_local_decl_count(0)
-  on_i32_const_expr(42 (0x2a))
-  on_return_expr
-  end_function_body(0)
-end_function_bodies_section
-end_module
+BeginModule(version: 1)
+  BeginTypeSection(5)
+    OnTypeCount(1)
+    OnType(index: 0, params: [], results: [i32])
+  EndTypeSection
+  BeginFunctionSection(2)
+    OnFunctionCount(1)
+    OnFunction(index: 0, sig_index: 0)
+  EndFunctionSection
+  BeginExportSection(8)
+    OnExportCount(1)
+    OnExport(index: 0, kind: func, item_index: 0, name: "main")
+  EndExportSection
+  BeginCodeSection(7)
+    OnFunctionBodyCount(1)
+    BeginFunctionBody(0)
+    OnLocalDeclCount(0)
+    OnI32ConstExpr(42 (0x2a))
+    OnReturnExpr
+    EndFunctionBody(0)
+  EndCodeSection
+EndModule
```

Comment:
Yeah, I forgot. the BinaryReaderInterpreter did two passes over the binary before since you need to first validate all the modifications to the data and elem segments, then only apply them if all are valid. This turned out to be a pain for the new system, so I ended up using two vectors to store the modifications and updating them later.